updateManager/client.go

```diff
@@ -216,6 +219,43 @@ func (um *Client) UpdateToVersion(version string, updateCompleteCallback func(st
 	return nil
 }
 
+// RemoveAllVersionsExcept deletes all versions except for the specified version
+func (um *Client) RemoveAllVersionsExcept(omitVersion string) error {
+	var err error
+
+	root := um.Config.VersionsRoot()
+	if len(root) <= 1 {
+		logrus.WithField("versions-root", root).Fatal("Potentially dangerous versions-root configuration.")
+		return ErrRemovingAllVersions
+	}
+
+	err = afero.Walk(um.Fs, root, func(path string, info os.FileInfo, walkErr error) error {
```

Comment:
`Walk` recursively walks down the version root directory, I think a `ReadDir` maybe better here.

```golang
dirContent, err := afero.ReadDir(um.Fs, root)
if err != nil {
  // handle error
}
for _, info := range dirContent {
  if info.IsDir() && info.Name() != omitVersion {
    um.RemoveVersion(info.Name())
  }
}
```

Reply:
I misread the docs 😳